exporters/jaeger/include/opentelemetry/exporters/jaeger/recordable.h

```diff
@@ -7,12 +7,46 @@
 #include <opentelemetry/sdk/trace/recordable.h>
 #include <opentelemetry/version.h>
 
+#if (defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
+     __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#  define JAEGER_IS_LITTLE_ENDIAN 1
+#elif defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
+    __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#  define JAEGER_IS_LITTLE_ENDIAN 0
+#elif defined(_WIN32)
+#  define JAEGER_IS_LITTLE_ENDIAN 1
+#else
+#  error "Endian detection needs to be set up for your compiler"
```

Comment:
how about using run-time determination if it's not set by compiler, something like:
```
    const int value { 0x01 };
    bool isLittleEndian = ( *(static_cast<const unsigned char *>(static_cast<const void *>(&value))) == 0x01);
```

Reply:
I'd like to avoid the runtime check as it is a static property known at compile time. I think our current definition should cover most real cases if not all.

Reply:
For a reference implementation see https://github.com/abseil/abseil-cpp/blob/master/absl/base/internal/endian.h I think you can use that code (not directly if avoiding dependency is needed) but is a good implementation.